common/membership/ringpop/factory.go

```diff
@@ -52,53 +53,43 @@ const (
 	persistenceOperationTimeout = 10 * time.Second
 )
 
+type factoryParams struct {
+	fx.In
+	Config          *config.Membership
+	ServiceName     primitives.ServiceName
+	ServicePortMap  config.ServicePortMap
+	Logger          log.Logger
+	MetadataManager persistence.ClusterMetadataManager
+	RPCConfig       *config.RPC
+	TLSFactory      encryption.TLSConfigProvider
+	DCCollection    *dynamicconfig.Collection
+}
+
 // factory provides a Monitor
 type factory struct {
-	config         *config.Membership
-	channel        *tchannel.Channel
-	serviceName    primitives.ServiceName
-	servicePortMap map[primitives.ServiceName]int
-	logger         log.Logger
-
-	monitor         *monitor
-	metadataManager persistence.ClusterMetadataManager
-	rpcConfig       *config.RPC
-	tlsFactory      encryption.TLSConfigProvider
-	dc              *dynamicconfig.Collection
+	*factoryParams
```

Comment:
does it work if you embed it as a value instead of a pointer?

Reply:
Ok, I'm convinced; will change to a value.